app/(main)/diagnostics.tsx

```
@@ -1,14 +1,332 @@
+    anemiaRisk: 2,
+    thrombosisRisk: 1,
+    menstrualFlow: 2,
+  },
+  {
+    recordedAt: "2025-10-31T10:00:00Z",
+    anemiaRisk: 1,
+    thrombosisRisk: 2,
+    menstrualFlow: 1,
+  },
+  {
+    recordedAt: "2025-11-01T10:00:00Z",
+    anemiaRisk: 0,
+    thrombosisRisk: 1,
+    menstrualFlow: 0,
+  },
+];
+
+type DiagnosticsScreenProps = {
+  history?: RiskHistoryPoint[];
+};
+
+export default function DiagnosticsScreen({
+  history: historyProp,
+}: DiagnosticsScreenProps) {
+  const { token, userId } = useAuth();
+
+  const [history, setHistory] = useState<RiskHistoryPoint[]>(historyProp ?? []);
+  const [loading, setLoading] = useState(!historyProp);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (historyProp) return; // Don't fetch if history is passed as a prop
+
+    if (!token || !userId) {
+      // Not logged in, but for development purposes, show mock data.
+      setHistory(mockHistory);
+      setLoading(false);
+      return;
+    }
+
+    const load = async () => {
+      try {
+        setLoading(true);
+        setError(null);
+        const data = await getRiskHistory(token, userId);
+        if (data && data.length > 0) {
+          setHistory(data);
+        } else {
+          // If API returns no data, fallback to mock data for development
+          setHistory(mockHistory);
+        }
+      } catch (err: unknown) {
+        console.error("Failed to load risk history", err);
+
+        // Fallback to mock data if API fails, as requested for development
+        setHistory(mockHistory);
+
+        if (isAxiosError(err)) {
+          const status = err.response?.status;
+          if (status === 401) {
+            setError("Your session has expired. Please log in again.");
+          } else {
+            // Don't show a "not found" error, just use mock data for now.
+            setError(null);
+          }
+        } else {
+          setError(null); // Or a generic error message
```

This seems weird, correct me if I am wrong, but right now, for everything that isn't error 401, we just log the error, silently show mock data, and ERASE the error?
That way the user does not know anything about the error, so I think we should show some message sayign "we couldn't load your data."